BaragonCore/src/main/java/com/hubspot/baragon/models/BaragonSource.java

```diff
@@ -0,0 +1,51 @@
+package com.hubspot.baragon.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Optional;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class BaragonSource {
+  private final String name;
+  private final Optional<Integer> port;
+
+  @JsonCreator
```

Comment:
I believe that migration is required because existing JSON object stores sources as an array of strings `{ ... , "sources":["source1"]}` and after migration it would be array of objects `{ .... , "sources":[{"name":"source1","port":10}]}`.